src/DotVVM.Framework/Resources/Scripts/DotVVM.PostBackHandlers.ts

```diff
@@ -15,8 +25,21 @@ class ConfirmPostBackHandler extends DotvvmPostBackHandler {
     }
 }
 
-class DotvvmPostBackHandlers {
-    public confirm = (options: any) => new ConfirmPostBackHandler(<string>options.message);
+class PostbackOptions {
+    constructor(public readonly postbackId: number, public readonly sender?: HTMLElement, public readonly args : any[] = [], public readonly viewModel?: any, public readonly viewModelName?: string, public readonly validationTargetPath?: any) {}
+}
+
+class ConfirmPostBackHandler2 implements DotvvmPostbackHandler2 {
```

Comment:
Unfortunately, the postback handler api is documented and I don't see any way how to make it backward compatible, so I have created the second interface, although I don't like the solution very much. Maybe we could break the compatibility, but it'll result only in a runtime error if you don't use typescript (and upgrade the definitions...).

And I use `type` instead of `interface` because as far as I know, the type is more powerful - I don't think you can do the discriminated union in an interface. But I'm not a Typescript expert, do you think it would better to use interface?